### ISSUE TITLE
_queue.Empty 问题的另一个注意事项【已解决】

### ISSUE THREAD

**Issue by XuJianzhi** [Holasyb918/HeyGem-Linux-Python-Hack#60]
在把所有相关issue都看遍之后，完全照做，还有 _queue.Empty 的问题。
共卡了我6.5小时。废了很多脑细胞。
然后将sleep10秒改成sleep20秒。。。。就好了。。。。。。。。。。。。。。。。。。。
。。。。。。。。。。。。。。。。
卡了我六个半小时啊！

------------------------------

问个外行问题，，，我不太懂视频相关工作哈，，，为什么非要multiprocess呢？为何不直接运行呢？是因为原始开源工程是这样做的，所以沿用下来了？multiprocess能提升gpu的使用率？

**Comment by XuJianzhi** [comment]
另外留下我跑通的配置：

`
(env_heygem) root@job-4042-1749018727-jwgmc:/data/try/HeyGem-Linux-Python-Hack# cat environment_use.yml 
name: env_heygem
channels:
  - pytorch
  - https://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/main
  - https://mirrors.tuna.tsinghua.edu.cn/anaconda/cloud/pytorch/
  - https://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/main/
  - https://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/free/
  - defaults
dependencies:
  - _libgcc_mutex=0.1=main
  - _openmp_mutex=5.1=1_gnu
  - blas=1.0=mkl
  - brotli-python=1.0.9=py38h6a678d5_8
  - ca-certificates=2025.2.25=h06a4308_0
  - cudatoolkit=11.8.0=h6a678d5_0
  - gmp=6.3.0=h6a678d5_0
  - gmpy2=2.1.2=py38heeb90bb_0
  - intel-openmp=2023.1.0=hdb19cb5_46306
  - jpeg=9e=h5eee18b_3
  - lcms2=2.16=hb9589c4_0
  - ld_impl_linux-64=2.40=h12ee557_0
  - lerc=4.0.0=h6a678d5_0
  - libdeflate=1.22=h5eee18b_0
  - libffi=3.4.4=h6a678d5_1
  - libgcc-ng=11.2.0=h1234567_1
  - libgomp=11.2.0=h1234567_1
  - libjpeg-turbo=2.0.0=h9bf148f_0
  - libpng=1.6.39=h5eee18b_0
  - libstdcxx-ng=11.2.0=h1234567_1
  - libtiff=4.5.1=hffd6297_1
  - libwebp-base=1.3.2=h5eee18b_1
  - libxcb=1.17.0=h9b100fa_0
  - llvm-openmp=14.0.6=h9e868ea_0
  - lz4-c=1.9.4=h6a678d5_1
  - mkl=2023.1.0=h213fc3f_46344
  - mkl-service=2.4.0=py38h5eee18b_1
  - mkl_fft=1.3.8=py38h5eee18b_0
  - mkl_random=1.2.4=py38hdb19cb5_0
  - mpc=1.3.1=h5eee18b_0
  - mpfr=4.2.1=h5eee18b_0
  - mpmath=1.3.0=py38h06a4308_0
  - ncurses=6.4=h6a678d5_0
  - openjpeg=2.5.2=he7f1fd0_0
  - openssl=3.0.16=h5eee18b_0
  - pip=24.2=py38h06a4308_0
  - pthread-stubs=0.3=h0ce48e5_1
  - pysocks=1.7.1=py38h06a4308_0
  - python=3.8.20=he870216_0
  - pytorch-mutex=1.0=cpu
  - readline=8.2=h5eee18b_0
  - setuptools=75.1.0=py38h06a4308_0
  - sqlite=3.45.3=h5eee18b_0
  - sympy=1.13.3=py38h06a4308_0
  - tbb=2021.8.0=hdb19cb5_0
  - tk=8.6.14=h993c535_1
  - wheel=0.44.0=py38h06a4308_0
  - xorg-libx11=1.8.12=h9b100fa_1
  - xorg-libxau=1.0.12=h9b100fa_0
  - xorg-libxdmcp=1.1.5=h9b100fa_0
  - xorg-xorgproto=2024.1=h5eee18b_1
  - xz=5.6.4=h5eee18b_1
  - yaml=0.2.5=h7b6447c_0
  - zlib=1.2.13=h5eee18b_1
  - zstd=1.5.6=hc292b87_0
  - pip:
      - appdirs==1.4.4
      - apstone==0.0.8
      - audioread==2.1.9
      - blinker==1.8.2
      - boto3==1.37.38
      - botocore==1.37.38
      - build==1.2.2.post1
      - cachecontrol==0.14.2
      - certifi==2025.4.26
      - cffi==1.15.0
      - charset-normalizer==2.0.12
      - cleo==2.1.0
      - click==8.1.3
      - colorama==0.4.4
      - coloredlogs==15.0.1
      - crashtest==0.4.1
      - cryptography==45.0.3
      - cv2box==0.5.9
      - cycler==0.11.0
      - decorator==5.1.1
      - distlib==0.3.9
      - dulwich==0.21.7
      - einops==0.8.1
      - fastjsonschema==2.21.1
      - filelock==3.7.1
      - flask==3.0.3
      - flatbuffers==2.0
      - fonttools==4.36.0
      - freetype-py==2.3.0
      - fsspec==2025.3.0
      - huggingface-hub==0.0.8
      - humanfriendly==10.0
      - idna==3.3
      - imageio==2.19.3
      - importlib-metadata==4.11.4
      - importlib-resources==6.4.5
      - installer==0.7.0
      - itsdangerous==2.2.0
      - jaraco-classes==3.4.0
      - jeepney==0.9.0
      - jinja2==3.1.6
      - jmespath==1.0.1
      - joblib==1.1.0
      - keyring==24.3.1
      - kiwisolver==1.4.4
      - kornia==0.6.6
      - librosa==0.8.1
      - llvmlite==0.38.1
      - markupsafe==2.1.5
      - matplotlib==3.5.3
      - more-itertools==10.5.0
      - msgpack==1.1.0
      - networkx==2.6.3
      - numba==0.55.2
      - numexpr==2.8.6
      - numpy==1.22.4
      - onnxruntime-gpu==1.16.0
      - opencv-python==4.7.0.72
      - packaging==21.3
      - pexpect==4.9.0
      - pillow==9.1.1
      - pkginfo==1.12.1.2
      - platformdirs==4.3.6
      - poetry==1.8.5
      - poetry-core==1.9.1
      - poetry-plugin-export==1.8.0
      - pooch==1.6.0
      - protobuf==4.21.5
      - psutil==5.9.1
      - ptyprocess==0.7.0
      - pycparser==2.21
      - pyglet==1.5.26
      - pyopengl==3.1.0
      - pyparsing==3.0.9
      - pyproject-hooks==1.2.0
      - pyrender==0.1.45
      - python-dateutil==2.8.2
      - pywavelets==1.3.0
      - pyyaml==6.0
      - rapidfuzz==3.9.7
      - regex==2024.11.6
      - requests==2.27.1
      - requests-toolbelt==1.0.0
      - resampy==0.2.2
      - s3transfer==0.11.5
      - sacremoses==0.1.1
      - scikit-image==0.19.3
      - scikit-learn==1.0.2
      - scipy==1.10.1
      - secretstorage==3.3.3
      - sentencepiece==0.2.0
      - shellingham==1.5.4
      - six==1.16.0
      - soundfile==0.13.1
      - threadpoolctl==3.1.0
      - tifffile==2023.7.10
      - tokenizers==0.13.3
      - tomli==2.2.1
      - tomlkit==0.13.2
      - torch==2.1.2+cu118
      - torchaudio==2.1.2+cu118
      - torchvision==0.16.2+cu118
      - tqdm==4.67.1
      - transformers==2.3.0
      - trimesh==4.6.10
      - triton==2.1.0
      - trove-classifiers==2025.5.9.12
      - typeguard==2.13.3
      - typing-extensions==4.13.2
      - urllib3==1.26.9
      - virtualenv==20.31.2
      - werkzeug==3.0.6
      - zipp==3.20.2
prefix: /root/miniconda3/envs/env_heygem
`

**Comment by Holasyb918** [comment]
> 在把所有相关issue都看遍之后，完全照做，还有 _queue.Empty 的问题。 共卡了我6.5小时。废了很多脑细胞。 然后将sleep10秒改成sleep20秒。。。。就好了。。。。。。。。。。。。。。。。。。。 。。。。。。。。。。。。。。。。 卡了我六个半小时啊！
> 
> 问个外行问题，，，我不太懂视频相关工作哈，，，为什么非要multiprocess呢？为何不直接运行呢？是因为原始开源工程是这样做的，所以沿用下来了？multiprocess能提升gpu的使用率？

是的，沿用的。

**Comment by XuJianzhi** [comment]
后续：又有了，这个参数调到600也还是报这个错。。。。额。。。。。

**Comment by Holasyb918** [comment]
> 后续：又有了，这个参数调到600也还是报这个错。。。。额。。。。。

盲猜环境被改了，不然不应该出这个问题。
